Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/photo_field.tsx

```diff
@@ -0,0 +1,653 @@
+import { css, StyleSheet } from 'aphrodite/no-important';
+import * as React from 'react';
+import { Transition } from 'react-transition-group';
+import { DisplaySize, HBoxLayout, HLine, Padding, VBoxLayout } from '../../..';
+
+/** Various the mode that the PhotoField can be displayed at. */
+export enum DisplayMode {
+
+  /** Only the photo is visible. */
+  Display,
+
+  /** The uploader is visible. */
+  Uploading
+}
+
+interface Properties {
+
+  /** Determines the size to render the component at. */
+  displaySize: DisplaySize;
+
+  /** Determines if the ChangePictureModal is visible or not. */
+  displayMode: DisplayMode;
+
+  /** The URL the image is located at. */
+  imageSource?: string;
+
+  /** Determines if the image can be changed or not. */
+  readonly?: boolean;
+
+  /** A value from 0 to 200 that determines how zoomed in the image will be. */
+  scaling: number;
+
+  /** Callback to hide or show the uploader. */
+  toggleUploader: () => void;
+
+  /** Callback to store the file and the scaling for the file. */
+  onUpload: (newFileLocation: string, scaling: number) => void;
+}
+
+/** Displays an account's profile image. */
+export class PhotoField extends React.Component<Properties, {}> {
+  public static readonly defaultProps = {
+    readonly: false
+  };
+
+  public render(): JSX.Element {
+    const boxStyle = (() => {
+      switch(this.props.displaySize) {
+        case DisplaySize.SMALL:
+          return PhotoField.STYLE.boxSmall;
+        case DisplaySize.MEDIUM:
+          return PhotoField.STYLE.boxMedium;
+        case DisplaySize.LARGE:
+          return PhotoField.STYLE.boxLarge;
+      }
+    })();
+    const cameraIconStyle = (() => {
+      if(this.props.readonly) {
+        return PhotoField.STYLE.hidden;
+      } else {
+        return PhotoField.STYLE.cameraIcon;
+      }
+    })();
+    const imageSrc = (() => {
+      if(!this.props.imageSource) {
```

Comment:
Minor nitpick worth knowing... it's preferable to do:

```
if(x) {
  // consequent
} else {
  // default
}
```

Rather than:

```
if(!x) {
  // consequent
} else {
  // default
}
```